src/coreclr/nativeaot/Runtime/inc/CommonTypes.h

```diff
@@ -21,4 +43,6 @@ typedef uint32_t            UInt32_BOOL;    // windows 4-byte BOOL, 0 -> false,
 #define UInt32_FALSE        0
 #define UInt32_TRUE         1
 
+#define ABSTRACT_IMPL_NO_PURE_VIRTUAL { while(true); assert(!"abstract method"); }
```

Comment:
The assert will never be hit - it is after the infinite while loop.

I am surprised that this compiles. I would expect that the compiler will complain about missing return value or unreachable code.

Reply:
Should we keep the loop and delete assert?

Reply:
I think we want to have assert in case this gets called. I had the loop there just to stop compiler from complaining about missing return value.